tests/dragonfly/cluster_test.py

```diff
@@ -955,31 +955,14 @@ async def test_config_consistency(df_local_factory: DflyInstanceFactory):
       ]
     """
 
-    # push config only to source node
+    # Push config to the target node first, so that it won't reject the migration which will lead to errors
```

Comment:
we can not guarantee that config will be sent to the target node, because we can have 2 or more simultaneous migrations, so incoming and outgoing migration can be in the config

Reply:
As discussed, it's fine for the config to arrive in any order, but it won't _start_ until both sides received it.
I updated the C++ and test code as we discussed.